enhancements/microshift/microshift-apiserver-custom-certs.md

```diff
@@ -0,0 +1,188 @@
+1. Certifates will be passed to the tls-sni-cert-key command line flag preceding all the other certificates.
+1. kube-apiserver picks up the certificates and start serving them on the configured SNI.
+
+### Implementation Details
+
+The certs will be prepended to the []configv1.NamedCertificate list before the api server is started. (it will be added to `-tls-sni-cert-key` flag)
+
+this certification paths configuration and names will be prepended into the kube-apiserver `tls-sni-cert-key` command line flag.
+when same SNI appear in the CN part of the provided certs,this certificate will take precedence over the `default` external-signer. [ref](https://github.com/kubernetes/kubernetes/blob/98358b8ce11b0c1878ae7aa1482668cb7a0b0e23/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/named_certificates.go#L38)
+
+
+### API Extensions
+N/A
+
+### Risks and Mitigations
+* User provide incorrect/expired certificate.
+  kube-api server will continue serving with an expired cert - similiar approach is taken by OpenShift.
+  > users can mitigate this by using --insecure-skip-tls-verify client mode
+
+* User provided expired Certificate and Microshift service started/restarted 
+  Microshift will start with a warning in the logs,kube-api will continue serve with an expired cert - similiar approach is taken by OpenShift.
+  > users can mitigate this by using --insecure-skip-tls-verify client mode
+
+### Drawbacks
+* External certificate wont be automaticly renewed, so it requires manual Certificate rotation. 
+  similiar approach is taken by OpenShift.
+
+
+## Design Details
+N/A
+
+### Open Questions [optional]
+- How to handle kubeconfig generation for NamedCerts?
+
+### Test Plan
+add e2e test that will:
+- generate and sign certificates with custom ca.
+- change the default Microshift configuration to use the newly generated certs.
+- make sure system is functional using  generated external kubeconfig.
+
+### Graduation Criteria
+#### Dev Preview -> Tech Preview
+- Gather feedback from early adopters on possible issues.
+
+#### Tech Preview -> GA
+- Extensive testing with various certificates variations.
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+N/A
+
+### Version Skew Strategy
+N/A
+
+### Operational Aspects of API Extensions
+N/A
+
+#### Failure Modes
+the provided certs value will be validated before is it passed to the api-server flag
+
+this check will prevent Microshift service from starting:
+1. certificates files exists in the disk and readable by microshift process.
+1. certificates shouldnt override the internal local certificates which are managed internally.
+
+this check display warning message at the log and service will be started:
+1. certificates is expired.
+
+
+#### Support Procedures
+configured certs value are prepended to the  `tls-sni-cert-key`  TLS handshake command line flag  which is passed to the kube-apiserver:
```

Comment:
Configured certs values to the `tls-sni-cert-key` TLS handshake command line flag which...